Archives/VolFile.h

```diff
@@ -79,7 +88,8 @@ namespace Archives
 		};
 
 		uint32_t ReadTag(const std::array<char, 4>& tagName);
-		void WriteTag(StreamWriter& volWriter, uint32_t length, const char *tagText);
+		//void WriteTag(StreamWriter& volWriter, uint32_t length, const char *tagText);
+		void VolFile::WriteTag(StreamWriter& volWriter, uint32_t length, const std::array<char, 4>& tagName);
```

Comment:
You have a `VolFile::` prefix here.

Should also remove the old commented out code so it doesn't confuse people in the future.

Or maybe consider removing this function entirely, and just passing a constructed `StreamHeader` directly to `volWriter.Write`. It should be possible to even pass an unnamed temporary value. Though that may lose out on the nice exception message you provided in the function.

Reply:
On further review, I think this function should be removed entirely. The error reporting currently in write tag will only catch problems with the actual write process that are not specific to the tag. It would just catch if there was a generic write problem that happened to occur during the write process, if that makes sense. 

I should be better about deleting the class prefix in the header files. I tend to create the function first, then paste the prototype into the header file and get sloppy about cleaning it up.